bazel/envoy_binary.bzl

```diff
@@ -84,7 +91,7 @@ def _envoy_linkopts():
         "@envoy//bazel:boringssl_fips": [],
         "@envoy//bazel:windows_x86_64": [],
         "//conditions:default": ["-pie"],
-    }) + _envoy_select_exported_symbols(["-Wl,-E"])
+    }) + _envoy_select_exported_symbols(["-Wl,-E"]) + _envoy_select_exported_lua_symbols(["-Wl,--dynamic-list=/source/lua_symbols.ldsym"])
```

Comment:
note this flag differs on macOS so it would be nice to bake that into this logic somehow so those builds either ignore this, or actually do it correctly if you're targeting darwin